lib/build.js

```diff
@@ -152,24 +154,36 @@ class BuildModel extends BaseModel {
     start() {
         // Make sure that a pipeline and job is associated with the build
         return this.job
-            .then(job =>
-                job.pipeline.then(pipeline =>
+            .then((job) => {
+                let buildTimeout = DEFAULT_BUILD_TIMEOUT;
```

Comment:
can you move this logic to a separate function `const builtTimeout = getBuildTimeout(this[executor])` thereby hiding the special treatment given to kubernetes behind it.

Reply:
I added `getBuildTimeout()` method.